server/app/category_boards.go

```diff
@@ -122,29 +124,58 @@ func (a *App) createBoardsCategory(userID, teamID string, existingCategoryBoards
 		}
 
 		if !belongsToCategory {
-			if err := a.AddUpdateUserCategoryBoard(teamID, userID, map[string]string{board.ID: createdCategory.ID}); err != nil {
-				return nil, fmt.Errorf("createBoardsCategory failed to add category-less board to the default category, defaultCategoryID: %s, error: %w", createdCategory.ID, err)
+			boardIDsToAdd = append(boardIDsToAdd, board.ID)
+			newBoardMetadata := model.CategoryBoardMetadata{
+				BoardID: board.ID,
+				Hidden:  false,
 			}
+			createdCategoryBoards.BoardMetadata = append(createdCategoryBoards.BoardMetadata, newBoardMetadata)
+		}
+	}
 
-			createdCategoryBoards.BoardIDs = append(createdCategoryBoards.BoardIDs, board.ID)
+	if len(boardIDsToAdd) > 0 {
+		if err := a.AddUpdateUserCategoryBoard(teamID, userID, createdCategory.ID, boardIDsToAdd); err != nil {
+			return nil, fmt.Errorf("createBoardsCategory failed to add category-less board to the default category, defaultCategoryID: %s, error: %w", createdCategory.ID, err)
 		}
 	}
 
 	return createdCategoryBoards, nil
 }
 
-func (a *App) AddUpdateUserCategoryBoard(teamID, userID string, boardCategoryMapping map[string]string) error {
-	err := a.store.AddUpdateCategoryBoard(userID, boardCategoryMapping)
+func (a *App) AddUpdateUserCategoryBoard(teamID, userID, categoryID string, boardIDs []string) error {
+	if len(boardIDs) == 0 {
+		return nil
+	}
+
+	err := a.store.AddUpdateCategoryBoard(userID, categoryID, boardIDs)
 	if err != nil {
 		return err
 	}
 
-	wsPayload := make([]*model.BoardCategoryWebsocketData, len(boardCategoryMapping))
+	userCategoryBoards, err := a.GetUserCategoryBoards(userID, teamID)
+	if err != nil {
+		return err
+	}
+
+	var updatedCategory *model.CategoryBoards
+	for i := range userCategoryBoards {
+		if userCategoryBoards[i].ID == categoryID {
```

Comment:
@harshilsharma63 instead of fetching all userCategoryBoards here and then looping through lookign for a category ID, can we just call SqlStore.GetCategory(id)  ?